### PR TITLE
Chat server

### PR DESCRIPTION
**Pull request by Vicx95**

### CODE REVIEW

File: server/chatserver.cpp [hunk 1]
```diff
@@ -1,56 +1,41 @@
...
-                            QWebSocketServer::NonSecureMode,
-                            this)}
-{
-    if(m_webSocketServer->listen(QHostAddress::Any, port))
-    {
+      m_webSocketServer{std::make_unique<QWebSocketServer>("Chat Server", QWebSocketServer::NonSecureMode, this)} {
+    if (m_webSocketServer->listen(QHostAddress::Any, port)) {
         qInfo() << "Server listening on port " << port;
-        connect(m_webSocketServer, &QWebSocketServer::newConnection, this, &ChatServer::onNewConnection);
+        connect(&*m_webSocketServer, &QWebSocketServer::newConnection, this, &ChatServer::onNewConnection);
```
Comment (lukasz126) [inline server/chatserver.cpp:12]: please use m_webSocketServer.get()

File: server/chatserver.cpp [hunk 2]
```diff
@@ -1,56 +1,41 @@
...
-{
-    auto clientSocket = m_webSocketServer->nextPendingConnection();
-    qInfo() << getIdentifier(clientSocket) << " connected!";
-    clientSocket->setParent(this);
+ChatServer::~ChatServer() { m_webSocketServer->close(); }
 
-    connect(clientSocket, &QWebSocket::textMessageReceived, this, &ChatServer::processMessage);
-    connect(clientSocket, &QWebSocket::disconnected, this, &ChatServer::socketDisconnected);
+void ChatServer::onNewConnection() {
+    auto clientSocket = std::make_unique<QWebSocket*>(m_webSocketServer->nextPendingConnection());
```
Comment (lukasz126) [inline server/chatserver.cpp:19]: I think it is a bad idea to create unique_ptr for raw ptr,
maybe try like this:
auto clientSocket = std::make_unique<QWebSocket>(*m_webSocketServer->nextPendingConnection());
?

File: server/chatserver.cpp [hunk 3]
```diff
@@ -1,56 +1,41 @@
...
-static QString getIdentifier(QWebSocket* socket)
-{
-    return QStringLiteral("%1:%2").arg(socket->peerAddress().toString(),
-                                       QString::number(socket->peerPort()));
+static QString getIdentifier(QWebSocket* socket) {
+    return QStringLiteral("%1:%2").arg(socket->peerAddress().toString(), QString::number(socket->peerPort()));
 }
 
-ChatServer::ChatServer(quint16 port, QObject *parent)
+ChatServer::ChatServer(quint16 port, QObject* parent)
```
Comment (lukasz126) [inline server/chatserver.cpp:7]: const?

File: server/chatserver.h [hunk 4]
```diff
@@ -18,6 +18,6 @@ private slots:
     void socketDisconnected();
 
 private:
-    QWebSocketServer* m_webSocketServer;
-    QList<QWebSocket*> m_clients;
+    std::unique_ptr<QWebSocketServer> m_webSocketServer;
+    std::vector<std::unique_ptr<QWebSocket*>> m_clients;
```
Comment (lukasz126) [inline server/chatserver.h:22]: I think we shouldn't use std::vector<std::unique_ptr to raw pointers, this may cause memory problems

**Review by lukasz126 — COMMENTED** [review]
I think we shouldn't use :unique_ptr to raw pointers, this may cause memory problems

File: server/chatserver.cpp (hunk 2 above)
Comment (Vicx95) [reply server/chatserver.cpp:19]: It's not possible, when I do this I got chatserver.cpp:19: error: In template: call to deleted constructor of 'QWebSocket'

File: server/chatserver.h (hunk 4 above)
Comment (Vicx95) [reply server/chatserver.h:22]: Do you have other idea? We have to store QWebSocket*, there is no option to store as a normal type

**Comment by ziobron** [comment]
@Vicx95 Your solution has not passed all checks (maybe it was only a formatting-check?) and you haven't modified this Pull Request during the last 7 days. It will be automatically closed after the next 7 days if it remains untouched.

**Comment by ziobron** [comment]
@Vicx95 Your solution has not passed all checks (maybe it was only a formatting-check?) and you haven't modified this Pull Request during the last 7 days. It will be automatically closed after the next 7 days if it remains untouched.

**Comment by ziobron** [comment]
We are closing this PR. If you want to fix your code and get points, please re-open it or raise a new PR

**Pull request closed without merging**